core/src/main/java/de/flo56958/MineTinker/Listeners/PlayerListener.java

```diff
@@ -30,66 +31,121 @@
+        String beginning = name[0].toLowerCase();
+
+        switch (beginning) {
+            case "shield":
+            case "wooden":
+                if (Lists.getWoodPlanks().contains(repair.getType())) {
+                    eligible = true;
+                }
+                break;
+            case "stone":
+                if (repair.getType().equals(Material.COBBLESTONE) || repair.getType().equals(Material.STONE)) {
+                    eligible = true;
+                }
+                break;
+            case "shears":
+            case "iron":
+                if (repair.getType().equals(Material.IRON_INGOT)) {
+                    eligible = true;
+                }
+                break;
+            case "golden":
+                if (repair.getType().equals(Material.GOLD_INGOT)) {
+                    eligible = true;
+                }
+                break;
+            case "diamond":
+                if (repair.getType().equals(Material.DIAMOND)) {
+                    eligible = true;
+                }
+                break;
+            case "bow":
+            case "crossbow":
+            case "fishing":
+                if (repair.getType().equals(Material.STICK) || repair.getType().equals(Material.STRING)) {
+                    eligible = true;
+                }
+                break;
+            case "leather":
+                if (repair.getType().equals(Material.LEATHER)) {
+                    eligible = true;
+                }
+                break;
+            case "chainmail":
+                if (repair.getType().equals(Material.IRON_BARS)) {
+                    eligible = true;
+                }
+                break;
+            case "elytra":
+                if (repair.getType().equals(Material.PHANTOM_MEMBRANE)) {
+                    eligible = true;
+                }
+                break;
+            case "trident":
+                if (repair.getType().equals(Material.PRISMARINE_SHARD)) {
+                    eligible = true;
+                }
+                break;
+            case "turtle":
+                if (repair.getType().equals(Material.SCUTE)) {
+                    eligible = true;
+                }
+                break;
         }
 
         if (eligible) {
-            double dura = tool.getDurability();
-            double maxDura = tool.getType().getMaxDurability();
+            Damageable meta = (Damageable)tool;
+
+            int dura = meta.getDamage();
+            short maxDura = tool.getType().getMaxDurability();
             int amount = e.getWhoClicked().getItemOnCursor().getAmount();
-            double percent = Main.getPlugin().getConfig().getDouble("DurabilityPercentageRepair");
+            float percent = (float)Main.getPlugin().getConfig().getDouble("DurabilityPercentageRepair");
 
             while (amount > 0 && dura > 0) {
-                dura = dura - (maxDura * percent);
+                dura = (int)(dura - (maxDura * percent));
```

Comment:
Instead of casting I would do Math.round() since on the right side is now a float Math.round() returns a int